bin/varnishd/cache/cache_req_body.c

```diff
@@ -43,6 +43,47 @@
 #include "storage/storage.h"
 #include "hash/hash_slinger.h"
 
+/*----------------------------------------------------------------------
+ * Check and potentially update req framing headers.
+ */
+
+static ssize_t
+vrb_cached(struct req *req, ssize_t req_bodybytes)
+{
+	ssize_t l0, l;
+
+	CHECK_OBJ_NOTNULL(req, REQ_MAGIC);
+	AZ(req->req_body_cached);
+	assert(req_bodybytes >= 0);
+
+	l0 = http_GetContentLength(req->http0);
+	l = http_GetContentLength(req->http);
+
+	assert(req->req_body_status->avail > 0);
+	if (req->req_body_status->length_known) {
+		assert(req_bodybytes == l0);
+		assert(req_bodybytes == l);
+		AZ(http_GetHdr(req->http0, H_Transfer_Encoding, NULL));
+		AZ(http_GetHdr(req->http, H_Transfer_Encoding, NULL));
+	} else {
+		/* We must update also the "pristine" req.* copy */
+		AZ(http_GetHdr(req->http0, H_Content_Length, NULL));
+		assert(l0 < 0);
+		http_Unset(req->http0, H_Transfer_Encoding);
+		http_PrintfHeader(req->http0, "Content-Length: %ju",
```

Comment:
This problem existed before the proposed change:
I think this conflicts with [Req_Rollback()](https://github.com/varnishcache/varnish-cache/blob/142c3a012935ceb68ffd6fe753f5498eedd0c8ed/bin/varnishd/cache/cache_req.c#L220), because the header is created on the rolled back part of the client workspace.
Because the case is likely not particularly relevant and given that [a variant of the problem already exists](https://github.com/varnishcache/varnish-cache/blob/142c3a012935ceb68ffd6fe753f5498eedd0c8ed/bin/varnishd/cache/cache_req_body.c#L303), I think we might just veto rollback with a cached body.